src/terminal/parser/stateMachine.hpp

```diff
@@ -146,7 +146,14 @@ namespace Microsoft::Console::VirtualTerminal
 
         bool _isInAnsiMode;
 
-        std::wstring_view _run;
+        std::wstring_view _currentString;
+        size_t _runOffset;
+        size_t _runSize;
+
+        std::wstring_view _CurrentRun() const

```

Comment:
Maybe leave some commentary as to why this is the way that it is to prevent future refactoring that makes the perf worse.

Reply:
Commentary added.

Reply:
Thanks!